ingestion-beam/src/main/java/com/mozilla/telemetry/util/KeyStore.java

```diff
@@ -0,0 +1,104 @@
+package com.mozilla.telemetry.util;
+
+import com.google.api.pathtemplate.ValidationException;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.security.PrivateKey;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.io.IOUtils;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.loader.SchemaLoader;
+import org.jose4j.jwk.PublicJsonWebKey;
+import org.jose4j.lang.JoseException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+/**
+ * Store Private Keys derived from JSON Web Keys (JWK) decrypted via the GCP Key
+ * Management Service (KMS).
+ */
+public class KeyStore {
+
+  /**
+   * Fetch a key from the store.
+   *
+   * @param path fully qualified path to the key
+   * @return private key for decryption
+   */
+  public PrivateKey getKey(String path) {
+    ensureKeysLoaded();
+    PrivateKey key = keys.get(path);
+    return key;
+  }
+
+  private final String metadataLocation;
+  private transient Map<String, PrivateKey> keys;
+
+  public static KeyStore of(String metadataLocation) {
+    return new KeyStore(metadataLocation);
+  }
+
+  private KeyStore(String metadataLocation) {
+    this.metadataLocation = metadataLocation;
+  }
+
+  @VisibleForTesting
+  int numLoadedKeys() {
+    ensureKeysLoaded();
+    return keys.size();
+  }
+
+  private void loadAllKeys() throws IOException, ValidationException {
+    final Map<String, PrivateKey> tempKeys = new HashMap<>();
+
+    Schema schema;
+    try (InputStream inputStream = Resources.getResource("keystore-metadata.schema.json")
+        .openStream()) {
+      JSONObject schemaBytes = new JSONObject(new JSONTokener(inputStream));
```

Comment:
We should factor all json parsing logic into ingestion-beam's `Json.java`. In this case, this call would become `Json.readValue` and we'd add a new variant of that function that takes in an InputStream. That way, we're consistently using Jackson with the JsonOrgModule to create JSONObject instances.

Reply:
Trying to validate an ArrayNode is frustrating, it looks like there's a mismatch between the ArrayNodes and JSONArrays.

Reply:
I've factored this out, but there are some extra conversions needed to maintain the separation of libraries.